R/DataTypes.R

```diff
@@ -309,12 +309,19 @@ object_type <- function(obj) {
   if (is.factor(obj)) return("factor")
   if (is(obj, "POSIXct")) return("datetime")
   if (is(obj, "Date")) return("date")
-  if (is(obj, "blob")) return("binary")
+  if (is_blob(obj)) return("binary")
   if (is(obj, "difftime")) return("time")
 
   return(typeof(obj))
 }
 
+is_blob <- function(obj) {
+  if (is(obj, "blob")) return(TRUE)
+  if (is.object(obj) && any(class(obj) != "AsIs")) return(FALSE)
+  # Assuming raw inside naked lists, not checking
+  is.list(obj)
```

Comment:
Why do we need this assumption? Couldn't we check it with something like `is.list(x) && length(x) > 0 && is.raw(x[[1]])`

Reply:
Because checking `x[[1]]` is not enough, it may be `NULL`. In the worst case we'd iterate over the entire vector just for checking -- safer but slower. Happy to add that if you prefer.

Reply:
Added.